Add glossary page (#63)
ceramicnetwork/docs: docs/learn/glossary.md

### StreamID
A StreamID is an immutable identifier for a [stream](#stream). StreamIDs enable continuous streams of data to be referenced by a persistent identifier instead of by constantly-changing IPFS [CIDs](#cid).

### StreamTypes
StreamTypes are functions used for processing updates to [streams](#streams). StreamTypes handle everything from defining the data structure of the stream, to what can be stored in its [commits](#commits), its state transitition function, [authentication](#authentication) requirements, and [conflict resolution strategy](#conflict-resolution-strategy). Every stream must specify a StreamType; and StreamTypes run on Ceramic [nodes](#nodes). Ceramic comes [pre-installed with various StreamTypes](../streamtypes/overview.md), or you can code your own.

### Commits
Commits are individual [IPFS](#ipfs) records that make up a [stream](#streams). Streams may contain one or more commits.

### Genesis commit
A genesis commit is the first [commit](#commit) in a [stream](#streams).

### Signed commit

> Signed commits are [commits](#commits) that update the [state](#state) of a [stream](#streams).

Reviewer: Mention that these commits are cryptographically signed by a public key.

### Anchor commit

> Anchor commits are [commits](#commits) that contain a blockchain timestamp, providing an immutable record of time and ordering to other commits in the [stream](#streams).

Reviewer: ```suggestion
Anchor commits are [commits](#commits) that contain a blockchain timestamp, providing an immutable record of time and ordering to other commits in the [stream](#streams), sometimes known as a *proof-of-publication*.
```

### CommitID
A commitID is an immutable identifier for a specific [commit](#commits) in a [stream](#streams).

### State
State represents the state of a [stream](#streams) at various points in time. When a stream is loaded or queried from the [network](#networks), the current state is returned.

### Tip

> A tip is the most recent [commit](#commits) of a [stream](#streams).

Reviewer: ```suggestion
A tip is the CID for the most recent [commit(s)](#commits) of a [stream](#streams).
```

### DIDs
DIDs is the [W3C standard](https://www.w3.org/TR/did-core/) for decentralized identifiers. The DID specification outlines a standard URI scheme for creating a persistent decentralized identifier (DID) for a given subject as well as resolving metadata about that identifier via a [DID document](#did-document). DIDs are used as an [authentication](#authentication) mechanism by most [StreamTypes](#streamtypes).

### DID methods
DID methods are implementations of the [DID](#dids) specification. DID methods must specify a name for the method in the form of a string (see below), a description of where the [DID document](#did-document) is stored (or how it is statically generated), and a [DID resolver](#did-resolver) which can return a DID document given a URI that conforms to that particular DID method. There are over 40 DID methods on the W3C's official DID registry. Ceramic can support any DID method if needed, and currently supports the [3ID DID method](../authentication/dids/3id.md) and the [Key DID method](../authentication/dids/key.md). DID URIs look like this:

```

> did:<method-name>:<randomstring>

Reviewer: ```suggestion
did:<method-name>:<method-specific-identifier>
```

### Networks
Networks are collections of Ceramic [nodes](#nodes) that share specific configurations and communicate over a dedicated [libp2p](#libp2p) topic. Networks are discrete from one another. [Streams](#streams) that exist on one network are not discoverable on or portable to another. Currently, Ceramic has three primary networks: [mainnet](#mainnet), [Clay Testnet](#clay-testnet), and [dev unstable](#dev-unstable).

### Mainnet
Mainnet is the Ceramic [network](#networks) used for production deployments. For more information on mainnet, see the Networks page.

### Clay Testnet
Clay Testnet is a Ceramic [network](#networks) used for community testing and development purposes. For more information on Clay Testnet, see the Networks page.

### Dev Unstable

> Dev Unstable is a Ceramic [network](#networks) used by Ceramic core protocol developers for testing new protocol features and release candidates. It should be considered unstable and highly experimental.

Reviewer: We also test release candidates on Clay. On dev unstable we test the most recent commits on the `develop` branch of js-ceramic.

## Underlying technologies

### IPFS
[IPFS](https://ipfs.io) is the Interplanetary File System. Simply put, IPFS is a way to address static content using [CIDs](#cid) and to discover this content over a peer-to-peer network of nodes. Ceramic relies on IPFS for storing the [commits](#commits) that make up [streams](#streams) and discovering this data over the [network](#networks).

### CID

> A CID (content identifier) is an immutable identifier for a discrete piece of static content stored on [IPFS](#ipfs). CIDs are essentially a hash of the content along with metadata that describes how the content is encoded. Ceramic [streams](#streams) consist of multiple CIDs, encoded using [dag-jose](#dag-jose), and linked together using [IPLD](#ipld).

Reviewer: Commits are not only encoded using dag-jose, we also use dag-cbor, but other codecs could be used as well.

Reviewer: ```suggestion
A CID (content identifier) is an immutable identifier for a discrete piece of static content stored on [IPFS](#ipfs). CIDs are essentially a hash of the content along with metadata that describes how the content is encoded. Ceramic [streams](#streams) consist of multiple CIDs, encoded using [dag-jose](#dagjose), and linked together using [IPLD](#ipld).
```

### IPLD
[IPLD](https://ipld.io) (Interplanetary Linked Data) is the data structures layer of [IPFS](#ipfs). It is used to link multiple [CIDs](#cid) together into higher-level linked-data structures. Ceramic uses IPLD to create the data structures for [streams](#streams).

> ### Dag-Jose

Reviewer: ```suggestion
### DagJOSE
```

> Dag-Jose is a codec for [IPLD](#ipld) which stores content in [IPFS](#ipfs) using IETF's JOSE (JSON object signing and encryption) format. Each dag-jose object is actually two [IPLD](#ipld) objects: one to store the metadata and one to store the payload. This makes it safe for encryption without leaking unnecessary metadata.

Reviewer: ```suggestion
[DagJOSE](https://specs.ipld.io/block-layer/codecs/dag-jose.html) is a codec for [IPLD](#ipld) which stores content in [IPFS](#ipfs) using IETF's JOSE (JSON object signing and encryption) format.  It supports both *signed* and *encrypted* objects. JWS is used for *signed* objects and it encodes the payload as a CID, which means that the actualy payload is a separate [IPLD](#ipld) object. JWE is used for *encrypted* objects, it requires the ciphertext to be a CID, in order to not leak the full cleartext an *inline CID* is used to encode the entire cleartext inside of the CID. For more information refer to the [DagJOSE spec](https://specs.ipld.io/block-layer/codecs/dag-jose.html).
```

### Libp2p

> [Libp2p](https://libp2p.io) is the peer-to-peer networking protocol that is used by Ceramic. It is included as part of the [IPFS](#ipfs) stack. Ceramic relies on libp2p for discovering data over the [network](#networks) and communicating between [nodes](#nodes).

Reviewer: Mention that libp2p is also used by Eth2 and Polkadot?
